src/program/ps/ps.lua

```diff
@@ -14,26 +15,49 @@ end
 
 local function parse_args (args)
    local opt = {}
+   local preferpid = false
    function opt.h (arg) usage(0) end
-   args = lib.dogetopt(args, opt, "h", {help='h'})
+   function opt.p (arg) preferpid = true end
+   args = lib.dogetopt(args, opt, "h:p", {help='h', pid='p'})
    if #args ~= 0 then usage(1) end
+   return preferpid
+end
+
+local function appname_resolver()
+    local apps = {}
+    for name, pid in pairs(app.enumerate_named_programs()) do
+        apps[pid] = name
```

Comment:
Let's not use the name "app" here, as it also means the components of a flow network within snabb.  I suggest "instances", but many things can work